arch/arm/src/imxrt/imxrt_flexpwm.c

```diff
@@ -774,7 +806,7 @@ static int pwm_change_freq(struct pwm_lowerhalf_s *dev,
 {
   struct imxrt_flexpwm_s *priv = (struct imxrt_flexpwm_s *)dev;
 #ifdef CONFIG_PWM_MULTICHAN
-  uint8_t shift = info->channels[channel].channel - 1;
+  uint8_t shift = info->channels[channel].channel >> 1;
```

Comment:
Channel numbers are now from 0 to 7. MOD1 of FlexPWM has channel 0 for Output A and channel 1 for Output B. MOD2 has channels 2 and 3, etc. In order to get shift form 0 to 3, we need to divide channel number by 2, thus shifting left.

Reply:
Ok, I got it. In this case channel numbers should be 1 to 8 since number 0 means the channel is not used (and is skipped per line [1258](https://github.com/apache/nuttx/blob/c9424becbb7f6813b3576a7646f1bd51517adf41/arch/arm/src/imxrt/imxrt_flexpwm.c#L1258)) . Therefore
```
uint8_t shift = (info->channels[channel].channel - 1) >> 1;
```
should be used. Otherwise we get mixed channels A and B. We should have MOD1 has channel 1 for output A and channel 2 for output B etc.

Reply:
The problem I see now is that channel numbers should not start with 0, because this is reserved for not defined channel, I believe. At least I can see that in rgbled driver. Why is this not documented somehow in pwm.h?

Reply:
Oh yea, my bad. You are right.

Reply:
I have already corrected code.

Reply:
Have also changed the Summary.

Reply:
Thanks! I created PR #8791 adding this to pwm.h. You are correct it should have been documentated.